src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Program.cs

```diff
@@ -40,7 +40,19 @@
 
 static async Task RunAsync(ServerConfiguration serverConfiguration, CancellationToken cancellationToken)
 {
-    // Before we initialize the LSP server we can't send LSP log messages.
+    // Redirect Console.Out when using stdio to try prevent corruption of the LSP output
+    if (serverConfiguration.UseStdIo)
+    {
+        if (serverConfiguration.ServerPipeName is not null)
+        {
+            throw new Exception("Server cannot be started with both --stdio and --pipe options.");
+        }
+
+        // Redirect Console.Out to try prevent the standard output stream from being corrupted.
```

Comment:
to indicate this is intentionally done before the logger is setup
```suggestion
        // Redirect Console.Out to try prevent the standard output stream from being corrupted.  This should be done before the logger is as it can write to the standard output.
```